typescript_test/test.ts

```diff
@@ -0,0 +1,346 @@
+import {
+  createSelector,
+  defaultMemoize,
+  createSelectorCreator,
+  createStructuredSelector,
+  ParametricSelector,
+} from '../src/index';
+
+
+
+function testSelector() {
+  type State = {foo: string};
+
+  const selector = createSelector(
+    (state: State) => state.foo,
+    (foo) => foo,
+  );
+
+  const foo: string = selector({foo: 'bar'});
+
+  // typings:expect-error
+  selector({foo: 'bar'}, {prop: 'value'});
+
+  // typings:expect-error
+  const num: number = selector({foo: 'bar'});
+
+  // typings:expect-error
+  createSelector(
+    (state: {foo: string}) => state.foo,
+    (state: {bar: number}) => state.bar,
+    (foo, bar) => 1
+  );
+}
+
+type Component<P> = (props: P) => any;
+
+declare function connect<S, P, R>(selector: ParametricSelector<S, P, R>):
+  (component: Component<P & R>) => Component<P>;
+
+function testConnect() {
+  connect(
+    createSelector(
+      (state: {foo: string}) => state.foo,
+      foo => ({foo}),
+    )
+  )(props => {
+    // typings:expect-error
+    props.bar;
+
+    const foo: string = props.foo;
+  });
+
+  const connected = connect(
+    createSelector(
+      (state: {foo: string}) => state.foo,
+      (state, props: {bar: number}) => state.bar,
```

Comment:
I think this should be ```props.bar``` not ```state.bar```?

To avoid this issue it would be nice to turn on ```noImplicitAny``` for the tests and then specify type ```never``` for unused params? 

```
(state: never, props: {bar: number}) => props.bar,
```